### ISSUE TITLE
Unable to Assign Different Weights to Classes for Functional Model

### ISSUE THREAD

**Issue by Soontosh** [keras-team/keras#20564]
I'm struggling to assign weights to different classes for my functional Keras model. I've looked for solutions online and everything that I have tried has not worked. I have tried the following:
1) Assigning `class_weights` as an argument for `model.fit`. I convert my datasets to numpy iterators first as `tf.data.Dataset` does not work with `class_weight`
```python
history = model.fit(
    train_dataset.as_numpy_iterator(),
    validation_data=val_dataset.as_numpy_iterator(),
    epochs=epochs,
    steps_per_epoch=steps_per_epoch,
    validation_steps=validation_steps,
    class_weight={0: 0.1, 1: 5.0}
)
```
2) Mapping sample weights directly to my `tf.data.Dataset` object.
```python
def add_sample_weights(features, labels):
        class_weights = tf.constant([0.1, 5.0])
        sample_weights = tf.gather(class_weights, labels)
        return features, labels, sample_weights

train_dataset = train_dataset.shuffle(buffer_size=10).batch(4).repeat().map(add_sample_weights)
val_dataset = val_dataset.batch(4).repeat().map(add_sample_weights)

# ...

model.compile(optimizer=optimizer, loss='binary_crossentropy', metrics=['accuracy'], weighted_metrics=['accuracy'])
```

For context, my dataset contains an imbalanced split of 70 / 30, for classes 0 and 1 respectively. I've been assigning the 0 a class weight of 0.1 and class 1 a class weight of 5. To my understanding, this should mean that the model "prioritizes" class 1 over class 0. However, my model still classifies everything as 0. How can I resolve this and is there anything that I'm missing? 

Let me know if you need any additional information to help me with this issue.

**Comment by Soontosh** [comment]
Nevermind, seems that I had exploding gradients which was preventing the class weights from being used at all
